### ISSUE TITLE
False Negative: Parameter is Never Used when destructuring in parameters

### ISSUE THREAD

**Issue by OptimisticPeach** [intellij-rust/intellij-rust#5109]
## Environment

* **IntelliJ Rust plugin version:** 0.2.118.2171-193
* **Rust toolchain version:** rustc 1.43.0-nightly (564758c4c 2020-03-08)
* **IDE name and version:** CLion 2019.3.4
* **Operating system:** Windows

## Problem description
When destructuring in a function's arguments, I get a 'Parameter `ty` is never used'-type message, even though it is used. 

## Steps to reproduce
Here is the minimal example from exactly what my code has:
```rust
union BarVal {
    as_bool: bool
}

enum BarTy {
    Bool,
}

struct Bar {
    ty: BarTy,
    val: BarVal,
}

enum SimpleBar {
    Bool(bool)
}

unsafe fn foo(Bar {ty, val}: Bar) -> SimpleBar {
    match ty {
        BarTy::Bool => SimpleBar::Bool(val.as_bool)
    }
}
```
and here is the minimal example I've found to cause this:
```rust
struct Bar {
    ty: usize,
}

fn foo(Bar {ty}: Bar) -> usize {
    ty
}
```
This does not occur if `Bar` is a tuple struct instead of a named struct.
